src/openzaak/components/catalogi/admin/catalogus.py

```diff
@@ -80,17 +101,176 @@ class CatalogusAdmin(
+
+                iotypen = retrieve_iotypen(
+                    catalogus_pk, request.session["file_content"]
+                )
+                request.session["iotypen"] = iotypen
+
+                besluittypen = retrieve_besluittypen(
+                    catalogus_pk, request.session["file_content"]
+                )
+                request.session["besluittypen"] = besluittypen
+
+                context["catalogus_pk"] = catalogus_pk
+                catalogus = Catalogus.objects.get(pk=catalogus_pk)
+
+                if iotypen:
+                    iotype_forms = InformatieObjectTypeFormSet(
+                        initial=[{"new_instance": instance} for instance in iotypen],
+                        form_kwargs={
+                            "catalogus_pk": catalogus_pk,
+                            "labels": [
+                                str(catalogus) + " - " + i["omschrijving"]
+                                for i in iotypen
+                            ],
+                        },
+                        prefix="iotype",
+                    )
+                    context["iotype_forms"] = iotype_forms
+
+                if besluittypen:
+                    besluittype_forms = BesluitTypeFormSet(
+                        initial=[
+                            {"new_instance": instance}
+                            for instance, uuids in besluittypen
+                        ],
+                        form_kwargs={
+                            "catalogus_pk": catalogus_pk,
+                            "labels": [
+                                str(catalogus) + " - " + i["omschrijving"]
+                                for i, uuids in besluittypen
+                            ],
+                        },
+                        prefix="besluittype",
+                    )
+                    context["besluittype_forms"] = besluittype_forms
+
+                if besluittypen or iotypen:
+                    return TemplateResponse(
+                        request, f"admin/catalogi/select_existing_typen.html", context,
+                    )
+                else:
+                    try:
+                        with transaction.atomic():
+                            import_zaaktype_for_catalogus(
+                                catalogus_pk, request.session["file_content"], {}, {}
+                            )
+
+                        self.message_user(
+                            request,
+                            _("ZaakType successfully imported"),
+                            level=messages.SUCCESS,
+                        )
+                        return HttpResponseRedirect(
+                            reverse(f"admin:catalogi_catalogus_changelist")
+                        )
+                    except CommandError as exc:
+                        self.message_user(request, exc, level=messages.ERROR)
+        elif "_select" in request.POST:
```

Comment:
Hmm, can you split it up in 2 views?

And move this whole inline view "thing" to a nice `admin_views` file and `ImportZaaktypeStep1View` and `ImportZaaktypeStep2View` class. and make it a class while at it. In `get_urls` you can reference it like this: `self.admin_site.admin_view(ImportZaaktypeView.as_view()))`